iceoryx_hoofs/vocabulary/include/iox/expected.hpp

```diff
@@ -136,9 +136,8 @@ template <typename ErrorType>
 class IOX_NO_DISCARD expected<ErrorType> final : public FunctionalInterface<expected<ErrorType>, void, ErrorType>
 {
   public:
-    /// @brief default ctor is deleted since you have to clearly state if the
-    ///         expected contains a success value or an error value
-    expected() = delete;
+    /// @brief default ctor is creates an object which contains an expected value
+    expected() noexcept = default;
```

Comment:
IMHO the committee did a mistake by making the `expected` default constructible. Is there any practical reason why we would need this?

Reply:
@mossmaurice Can you go a little bit more into the details please and provide a code snippet or use case where you see a problem?